tests/downloads/qml/test_single_download.cpp

```diff
@@ -486,7 +492,8 @@ TestSingleDownload::testSetHashSuccess() {
 
     singleDownload->setHash("c4e5f7fcbcef75924b2abde2b2e75f3f");
 
-    QVERIFY(spy.ensureSignalEmitted());
+    QEXPECT_FAIL("", "Test is broken", Abort);
```

Comment:
Please open issues about the test failures, and include the URL in the comment, so that these can be resolved later. Also, do these failures only happen with newer Qt (5.13+ on alpine/etc), or also on 5.9 that's in UT now?

Reply:
I think these tests here are fundamentally broken/wrong, so it's not an issue of some update breaking it. I also feel like the changes done to these test make it worse (but make it compile). E.g. in this setHash test, I'm thinking it was intended that singleDownload->setHash calls an internal method of _down called setHash and then the test would make some sense. (currently the value is passed in the "constructor" when download() is called on singleDownload). But in the current form (with this PR) they don't make much sense to me at all...

Reply:
ping, any opinion what I should do with those tests?

Reply:
Well, if the tests are broken in some way, I would prefer they are either fixed to work, or at least an issue is opened describing the problem, and using a `QSKIP()` at the beginning of the test including the issue number, so that we can track and fix that problem in a separate branch. Adding a `QEXPECT_FAIL` in the middle of a test because something is causing the signal to not be emitted, seems the wrong way to go about it.

But also, if we just remove the `QEXPECT_FAIL` calls here, do the tests still run and pass fine on Qt 5.9 with the other changes?

Reply:
Added QSKIP and commented out the code that doesn't compile (it's a single statement each).

Reply:
OK. So, `QSKIP("This test is broken.");` is not really helpful, nor is commenting out the `EXPECT_CALL` line because the mock object is missing the macro to mock the function in question. However, after looking into this further, and trying to understand what the heck is going on here, I'd have to say the tests are inappropriate for what the feature it's trying to test does. So I think the best thing is to just remove all of the `testSetHash*` and `testSetAlgorithm*` methods from `TestSingleDownload` instead. The tests look like they were mostly copy-pasta and never actually run so nobody knew they didn't work, nor didn't test the feature they were meant to test.

So let's remove the tests, and we can perhaps look at writing proper tests for the feature later.

Reply:
Done